examples/tests/NavigationSampleTests.elm

```diff
@@ -0,0 +1,35 @@
+module NavigationSampleTests exposing (all)
+
+import Expect
+import NavigationSample exposing (program)
+import Test exposing (..)
+import Test.Html.Event as Event
+import Test.Html.Query exposing (..)
+import Test.Html.Selector exposing (..)
+import TestContext exposing (..)
+
+
+all : Test
+all =
+    describe "NavigationSample"
+        [ test "renders navigation history" <|
+            \() ->
+                NavigationSample.program
+                    |> start
+                    |> simulate (find [ class "bears" ]) Event.click
+                    |> simulate (find [ class "cats" ]) Event.click
+                    |> expectView
+                    |> find [ class "history" ]
+                    |> Expect.all
+                        [ has [ text "#bears" ]
+                        , has [ text "#cats" ]
+                        ]
+        , test "works for user initiated navigation" <|
+            \() ->
+                NavigationSample.program
+                    |> start
+                    |> navigate "#cats"
```

Comment:
See answer above, yes, it should correctly work with full urls, I've tried to catch all the edge cases by testing it against a real project, and adding more tests